src/main/kotlin/com/openlattice/transporter/services/DataTransporterService.kt

```diff
@@ -0,0 +1,137 @@
+        val pollTimer: Histogram = Histogram.build()
+                .namespace("transporter")
+                .name("poll_duration_seconds")
+                .help("time to do one database sync poll")
+                .register()
+    }
+
+    private lateinit var acm: AssemblerConnectionManager
+    private lateinit var transporter: HikariDataSource
+
+    private val transporterState = HazelcastMap.TRANSPORTER_DB_COLUMNS.getMap( hazelcastInstance )
+
+    private fun syncTable(et: EntityType): Optional<Future<*>> {
+        val prev = transporterState.putIfAbsent(et.id, TransporterColumnSet(emptyMap()))
+        return if (prev == null) {
+            val props = dataModelService.getPropertyTypes(et.properties)
+            Optional.of(
+                    transporterState.submitToKey(et.id, TransporterSynchronizeTableDefinitionEntryProcessor(props).init(acm))
+            )
+        } else {
+            Optional.empty()
+        }
+    }
+
+    override fun init(acm: AssemblerConnectionManager): Void? {
+        this.acm = acm
+        this.transporter = acm.connect("transporter")
+        executor.submit {
+            val entityTypes = dataModelService.entityTypes.toList()
+            logger.info("initializing DataTransporterService with {} types", entityTypes.size)
+            val tablesCreated = entityTypes.map { et -> this.syncTable(et) }
+                    .filter { it.isPresent }
+                    .map { it.get().get() }
+                    .count()
+            logger.info("synchronization finished with {} entity type tables updated", tablesCreated)
+        }
+
+        eventBus.register(this)
+        return null
+    }
+
+    private fun partitions(entitySetIds: Set<UUID>): Collection<Int> {
+        return partitionManager.getPartitionsByEntitySetId(entitySetIds).values.flatten().toSet()
+    }
+
+    fun pollOnce() {
+        val timer = pollTimer.startTimer()
+        val futures = this.transporterState.keys.map { entityTypeId ->
+            val relevantEntitySets = validEntitySets(entityTypeId)
+            val entitySetIds = relevantEntitySets.map { it.id }.toSet()
+            val partitions = partitions(entitySetIds)
+            val ft = transporterState.submitToKey(entityTypeId, TransporterPropagateDataEntryProcessor(relevantEntitySets, partitions).init(acm))
+            entitySetIds.count() to ft
+        }
+        val setsPolled = futures.map { it.first }.sum()
+        try {
+            futures.forEach { it.second.get() }
+        } finally {
+            val duration = timer.observeDuration()
+            logger.debug("Total poll duration time for {} entity sets in {} entity types: {} sec", setsPolled, futures.size, duration)
+        }
+    }
+
+    private fun validEntitySets(entityTypeId: UUID): Set<EntitySet> {
+        return entitySetService.getEntitySetsOfType(entityTypeId)
+                .filter { !it.isLinking && !it.flags.contains(EntitySetFlag.AUDIT) }.toSet()
```

Comment:
we *will* theoretically at some point need to be able to run graph queries on audit data, unless we change our approach for feeds.... 😬

Reply:
I'll put audits back in.